pkg/network/protocols/events/configuration.go

```diff
@@ -8,45 +8,194 @@
 package events
 
 import (
+	"math"
 	"os"
 	"sync"
 
 	manager "github.com/DataDog/ebpf-manager"
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/asm"
+	"github.com/cilium/ebpf/features"
 
 	ddebpf "github.com/DataDog/datadog-agent/pkg/ebpf"
+	"github.com/DataDog/datadog-agent/pkg/network/usm/utils"
 	"github.com/DataDog/datadog-agent/pkg/util/kernel"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
-// handlerByProtocol holds a temporary reference to a `ddebpf.PerfHandler`
-// instance for a given protocol. This is done to simplify the usage of this
-// package a little bit, so a call to `events.Configure` can be later linked
-// to a call to `events.NewConsumer` without the need to explicitly propagate
-// any values. The map is guarded by `handlerMux`.
-var handlerByProtocol map[string]*ddebpf.PerfHandler
-var handlerMux sync.Mutex
+// defaultPerfBufferSize controls the amount of memory in bytes used *per CPU*
+// allocated for buffering perf event data
+var defaultPerfEventBufferSize = 16 * os.Getpagesize()
+
+// defaultPerfHandlerSize controls the size of the go channel that buffers perf
+// events (*ddebpf.PerfHandler). All perf events handled by this library have
+// fixed size (sizeof(batch_data_t)) which is ~4KB, so by choosing a value of
+// 100 we'll be buffering up to ~400KB of data in *Go* heap memory.
+const defaultPerfHandlerSize = 100
 
-// Configure event processing
-// Must be called *before* manager.InitWithOptions
+// Configure a given `*manager.Manager` for event processing
+// This essentially instantiates the perf map/ring buffers and configure the
+// eBPF maps where events are enqueued.
+// Note this must be called *before* manager.InitWithOptions
 func Configure(proto string, m *manager.Manager, o *manager.Options) {
-	setupPerfMap(proto, m)
+	if alreadySetUp(proto, m) {
+		return
+	}
+
 	onlineCPUs, err := kernel.PossibleCPUs()
 	if err != nil {
 		onlineCPUs = 96
 		log.Error("unable to detect number of CPUs. assuming 96 cores")
 	}
 
+	configureBatchMaps(proto, o, onlineCPUs)
+
+	// TODO: add a feature flag so we can optionally disable it
+	useRingBuffer := features.HaveMapType(ebpf.RingBuf) == nil
```

Comment:
please open a jira ticket to follow up on the TODO

Reply:
sounds good to me

Reply:
Done